server/apps/research/models/article.py

```
@@ -143,8 +144,17 @@ def save(self, *args, **kwargs):
             self.status = 'ready'
-
-        super().save(*args, **kwargs)
+        
+        if self.thumb and not hasattr(self.thumb, 'public_id'):
+            super().save(*args, **kwargs)
+        elif self.thumb and hasattr(self.thumb, 'public_id'):
+            try:
+                if not self.thumb.public_id:
+                    raise ValidationError("Failed to upload image to Cloudinary")
+            except Exception as e:
+                raise ValidationError(f"Image upload failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)